applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -4,3 +4,240 @@
+static struct bt_audio_stream streams[CONFIG_BT_AUDIO_BROADCAST_SNK_STREAM_COUNT];
+
+/* Mandatory support preset for both source and sink */
+static struct bt_audio_lc3_preset lc3_preset = BT_AUDIO_LC3_BROADCAST_PRESET_48_4_1;
+
+/* Create a mask for the maximum BIS we can sync to using the number of streams
+ * we have. We add an additional 1 since the bis indexes start from 1 and not
+ * 0.
+ */
+static const uint32_t bis_index_mask = BIT_MASK(ARRAY_SIZE(streams) + 1U);
+static uint32_t bis_index_bitfield;
+
+static void stream_started_cb(struct bt_audio_stream *stream)
+{
+	int ret;
+	struct event_t event;
+
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_STREAMING;
+	
+	ret = ctrl_events_put(&event);
+	ERR_CHK(ret);
+	
+	LOG_INF("Stream started");
+}
+
+static void stream_stopped_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Stream stopped");
+}
+
+static void stream_recv_cb(struct bt_audio_stream *stream, struct net_buf *buf)
+{
+	static uint32_t recv_cnt;
+	static uint32_t artificial_ts = 100;
+
+	if (receive_cb == NULL) {
+		LOG_ERR("The RX callback has not been set");
+		return;
+	}
+
+	receive_cb(buf->data, buf->len, false, artificial_ts);
+
+	artificial_ts += 10000;
+
+	recv_cnt++;
+	if ((recv_cnt % 1000U) == 0U) {
+		LOG_INF("Received %u total ISO packets", recv_cnt);
+	}
+}
+
+static struct bt_audio_stream_ops stream_ops = { .started = stream_started_cb,
+						 .stopped = stream_stopped_cb,
+						 .recv = stream_recv_cb };
+
+static bool scan_recv_cb(const struct bt_le_scan_recv_info *info, uint32_t broadcast_id)
+{
+	LOG_INF("Broadcast source found, waiting for PA sync");
+
+	return true;
+}
+
+static void scan_term_cb(int err)
+{
+	if (err) {
+		LOG_ERR("Scan terminated with error: %d", err);
+	}
+}
+
+static void pa_synced_cb(struct bt_audio_broadcast_sink *sink, struct bt_le_per_adv_sync *sync,
+			 uint32_t broadcast_id)
+{
+	if (broadcast_sink != NULL) {
+		LOG_ERR("Unexpected PA sync");
+		return;
+	}
+
+	LOG_INF("PA synced for broadcast sink with broadcast ID 0x%06X", broadcast_id);
+
+	broadcast_sink = sink;
+
+	LOG_INF("Broadcast source PA synced, waiting for BASE");
+}
+
+static void pa_sync_lost_cb(struct bt_audio_broadcast_sink *sink)
+{
+	if (broadcast_sink == NULL) {
+		LOG_ERR("Unexpected PA sync lost");
+		return;
+	}
+
+	LOG_WRN("Sink disconnected");
+
+	broadcast_sink = NULL;
+
+	// TODO: Must reset connection, this is not implemented yet
+}
+
+static void base_recv_cb(struct bt_audio_broadcast_sink *sink, const struct bt_audio_base *base)
+{
+	uint32_t base_bis_index_bitfield = 0U;
+
+	if (synced_to_broadcast) {
+		return;
+	}
+
+	LOG_INF("Received BASE with %u subgroups from broadcast sink", base->subgroup_count);
+
+	for (size_t i = 0U; i < base->subgroup_count; i++) {
+		for (size_t j = 0U; j < base->subgroups[i].bis_count; j++) {
+			const uint8_t index = base->subgroups[i].bis_data[j].index;
+
+			base_bis_index_bitfield |= BIT(index);
```

Comment:
I believe this is where we would print the codec settings received from source? base->subgroups[i].codec

Reply:
Yes, it looks like we can extract the codec settings in base_recv_cb()